src/redux/checkin.js

```diff
@@ -124,26 +124,110 @@ export const addBlocker = ({
   },
 });
 
+export const nextCheckinStep = () => ({
+  type: 'checkin/NEXT_CHECKIN_STEP',
+});
+
 // Selectors
+const getUserNameFromStateById = state => pipe(joinOn(state.users), prop('name'));
+
 export const getUsersList = pipe(prop('users'), Object.values);
 
-export const getTeamsList = (state) =>
-  Object.values(state.teams).map((team) => ({
+export const getTeamsList = state =>
+  Object.values(state.teams).map(team => ({
     id: team.id,
     name: team.name,
-    users: team.users.map(pipe(joinOn(state.users), prop('name'))),
+    users: team.users.map(getUserNameFromStateById(state)),
   }));
 
+const getChronologicalCheckins = (predicate, state) => {
+  return state.checkins.filter(predicate).sort((aCheckin, bCheckin) => {
+    if (aCheckin.createdAt < bCheckin.createdAt) {
+      return -1;
+    } else if (aCheckin.createdAt > bCheckin.createdAt) {
+      return 1;
+    }
+
+    return 0;
+  });
+};
+
+const getChronologicalCheckinsForTeam = (teamId, state) =>
+  getChronologicalCheckins(checkin => checkin.teamId === teamId, state);
+
+const getChronologicalCheckinsForUser = (userId, state) =>
+  getChronologicalCheckins(checkin => checkin.userId === userId, state);
+
+export const getTeamCheckinSummary = teamId => state => {
+  if (!state.teams[teamId]) {
+    return {};
+  }
+
+  const team = state.teams[teamId];
+  return {
+    id: team.id,
+    name: team.name,
+    checkins: getChronologicalCheckinsForTeam(teamId, state).map(checkin => ({
+      id: checkin.id,
+      user: getUserNameFromStateById(state)(checkin.userId),
+      createdAt: checkin.createdAt,
+      tasks: checkin.tasks,
+      blockers: checkin.blockers,
+    })),
+  };
+};
+
+const exportTaskForView = ({ id, description, completed }) => ({
```

Comment:
A couple opportunities to reduce duplication but this is pretty OK for the most part.

Reply:
Switch to use a curried `pick` function